internal/cli/login.go

```diff
@@ -141,3 +185,41 @@ func RunLogin(ctx context.Context, cli *cli, expired bool) (Tenant, error) {
 
 	return tenant, nil
 }
+
+// RunLoginAsMachine facilitates the authentication process using client credentials (client ID, client secret)
+func RunLoginAsMachine(ctx context.Context, inputs LoginInputs, cli *cli, cmd *cobra.Command) error {
+	if err := loginTenantDomain.Ask(cmd, &inputs.Domain, nil); err != nil {
+		return err
+	}
+
+	if err := tenantClientID.Ask(cmd, &inputs.ClientID, nil); err != nil {
+		return err
+	}
+
+	if err := tenantClientSecret.AskPassword(cmd, &inputs.ClientSecret, nil); err != nil {
```

Comment:
If this is intended for machine usage (automation scenarios where stdout won't be connected to a TTY), maybe we don't need to ask for the values?

Reply:
`--as-machine` flag has been removed so resolving this.